### ISSUE TITLE
Open files in same window of SPEK

### ISSUE THREAD

**Issue by todehefa** [alexkay/spek#144]
Hi guys!
I want to open the files in the same instance of Spek (same window)
At this moment it will open in another window every time. The only way is to drag and drop in the same window.
Please help :D !

**Comment by withmorten** [comment]
Cannot replicate this at all with Spek 0.8.2 windows build from the official site ... which Spek version are you using? OS?

**Comment by todehefa** [comment]
I'm using Spek 0.8.2 version on a Windows 10 machine.
If i go to an mp3 file for example and choose "open with" -> Spek (in this case) . it will open the mp3 files in a different window everytime. Open 5 mp3 files with spek ? Then i'll get 5 different windows (instances) of Spek.
The only way to open in the same window is to go to File-> Open -> Choose the file you want. It's very frustrating.
Why i want this to be implemented ? Because I want to check more files quickly (using a *ahk script for eg) but at this moment it will open a dozen of windows.
I hope i made myself clear (i'm not a native english speaker :D) and waiting for your help :)

**Comment by withmorten** [comment]
Oh that's what you mean!

I thought you meant when pressing Ctrl+O inside Spek.

No idea how to implement that, sorry ...

**Comment by todehefa** [comment]
or maybe it can be done by line comand in CMD. simulation the CTRL + O.
i didn't know it's that complicated to force an app to run as a single instance program... :( . (Mp3tag for e.g. run as that)

**Comment by withmorten** [comment]
It's probably not that complicated, I just don't know how to do it.

I'm sure a "Spek -i input.file" commandline parameter to open a file upon startup would be much easier to implement. I'll see if I get around to it.

**Comment by todehefa** [comment]
i found something on internet about how to run only one instance of application here: https://stackoverflow.com/questions/4191465/how-to-run-only-one-instance-of-application , but i have difficulties to implement this. 
You seem to be more experienced with programming than I do. (I have not coded for about 4 years now: D). Maybe you want to try or help me with this code.

**Comment by withmorten** [comment]
I'll have a look into this soon, time is short right now though.

I'm also not really that experienced either, but I might be able to do something. Worst case I'll just make a custom commandline parameter version for you or something.

**Comment by todehefa** [comment]
I can't believe it, i can't even compile it :)) it gives me a lot of errors. (i think i need some packages installed on my pc)

**Comment by withmorten** [comment]
https://github.com/withmorten/spek/blob/alternative/dist/win/README.md

Follow these instructions and it should work. You'll need a debian/ubuntu/linux Virtual Machine (or a "real" PC running them).

**Comment by todehefa** [comment]
I want to open the project in Visual Studio in order to make some changes to the code (try to make it single instance, or make a custom commandline parameter version). When i try to run the project a bunch of error pop out, that's because i don't have the necessary libraries i think.

What you sent me is a link to "Building the Windows installer" - and I have not yet reached this step. For now, I must make it run in visual studio :D .

Thank again for your help and guidance :)

**Comment by withmorten** [comment]
No, the link is to completely build it for Windows from the source, but on Linux.

Everything up to and including `./dist/win/bundle.sh` is for building the Spek.exe. `bundle.bat` is for packing the msi installer from the zip.

**Comment by todehefa** [comment]
so how do you managed to compile the code ? :)

**Comment by todehefa** [comment]
after the : "make pthreads ffmpeg wxwidgets -j1 JOBS=4" comand this is what i get:

> Missing requirement: autoconf
> Missing requirement: automake
> Missing requirement: autopoint
> Missing requirement: bison
> Missing requirement: flex
> Missing requirement: gperf
> Missing requirement: intltoolize
> Missing requirement: libtool
> Missing requirement: libtoolize
> Missing requirement: ruby
> Missing requirement: scons
> Missing requirement: 7za
> Missing requirement: gdk-pixbuf-csource
> bash: autoconf: command not found
> Wrong version of requirement: autoconf
> bash: automake: command not found
> Wrong version of requirement: automake
> 
> Please have a look at "docs/index.html" to ensure
> that your system meets all requirements.

**Comment by withmorten** [comment]
Hey, can you send me an email so we can chat there? I feel like we've cluttered up this specific issue enough with unrelated comments:

morten.with@gmail.com

**Comment by ranolfi** [comment]
I absolutely don't want this, since I usually use spek to compare between two or more audio files by using <kbd>Alt</kbd> + <kbd>Tab</kbd> to alternate between windows.
